wagon-providers/wagon-http-shared/src/main/java/org/apache/maven/wagon/shared/http/AbstractHttpClientWagon.java

```diff
@@ -487,6 +487,8 @@ private static CloseableHttpClient createClient()
             .disableConnectionState() //
             .setConnectionManager( httpClientConnectionManager ) //
             .setRetryHandler( createRetryHandler() )
+            .setServiceUnavailableRetryStrategy(
+                    new SimpleServiceUnavailableRetryStrategy( 10, 1000 ) )
```

Comment:
if we implement this it should be optional and retry number/time between retry configurable.